bindings/rust/s2n-tls/src/testing.rs

```diff
@@ -232,12 +263,16 @@ pub fn poll_tls_pair(mut pair: Pair<Harness, Harness>) -> Pair<Harness, Harness>
         }
     }
 
-    // TODO add assertions to make sure the handshake actually succeeded
+    #[cfg(test)]
+    {
+        assert!(pair.server.0.is_handshake_complete());
+        assert!(pair.client.0.is_handshake_complete());
+    }
```

Comment:
I'm not sure this is what the `TODO` meant. I think it would be better to call into the C code and ask what state it's in, rather than tracking it in Rust.

It also doesn't really seem to go with the PR changes :smile:

Reply:
Since we are checking for handshake success, I wanted a stronger check and it seems pretty trivial to add this. But since we want to inspect the C state its more complicated and I can remove it from this PR